server/task/src/main/scala/org/apache/james/task/eventsourcing/EventSourcingTaskManager.scala

```diff
@@ -62,6 +64,8 @@ class EventSourcingTaskManager @Inject @VisibleForTesting private[eventsourcing]
       workDispatcher),
     eventStore = eventStore)
 
+  listener.setEventSourcingSystem(eventSourcingSystem)
```

Comment:
how is it better than before?

Reply:
It's injected in a lot cleaner way in the next commit